봉사활동 게시판 관련 API 개발

LimNoah opened the pull request:
solved: #5

LimNoah committed:
conf: Modified application.yml.

Modified spring.jpa.datasource:url.

db-bongdari.cpijvq4ngvyo.ap-northeast-2.rds.amazonaws.com
->
jdbc:mysql://db-bongdari.cpijvq4ngvyo.ap-northeast-2.rds.amazonaws.com:3306/jwtest?user=admin&password=bongdari

for personal development environment.

LimNoah committed:
refactor: Update Board Entity.

Add relation with Member entity.

LimNoah committed:
feat: Create BoardDto.

Create BoardDto for use with getBoard API.

LimNoah committed:
feat: Create schema.sql.

Create schema.sql for init database tables.

LimNoah committed:
feat: Add getBoard API to BoardController.

Add getBoard API.

LimNoah committed:
refactor: Add field related to Member Entity.

Add member field related to Member Entity.

user ManyToOne Annotation.

LimNoah committed:
refactor: Update Member Entity.

LimNoah committed:
refactor: Refactor MemberRepository.

LimNoah committed:
test: Create test class.

Create BoardRepositoryTest and MemberRepositoryTest.

LimNoah committed:
feat: Create Exception Class.

Create ResourceNotFoundException class.

LimNoah committed:
config: Update application.yml.

Changed datasource url.

LimNoah committed:
feat: Add getBoardAPI.

Need to make service class and somethind to return.

LimNoah committed:
refactor: Update Board entity.

Add createdDate column.

LimNoah committed:
refactor: Update Board entity.

Add details / city columns.

LimNoah committed:
test: Update BoardRepositoryTest.

Add columns and test case.

LimNoah committed:
refactor: Update getBoard API.

Add service method to return object.

LimNoah committed:
refactor: Update ResourceNotFoundException.

LimNoah committed:
refactor: Update BoardResponseDto.

Add Builder annotation.

LimNoah committed:
refactor: Update BoardController.

Renamed service method name.

LimNoah committed:
feat: Create findById method.

LimNoah committed:
refactor: Update Board entity.

Modified createdDate column's datatype.

LocalDate -> LocalDateTime

LimNoah committed:
refactor: Refactor test code.

Delete createDate column test use assertThat.

LimNoah committed:
refactor: Refactor BoardRepository.

Add and remove QueryMethod.
-Add
	- findByCategoryAndCity
	- findByCategoryAndCityAndGu
-Remove
	- findByCategory

LimNoah committed:
refactor: Refactor BoardRepositoryTest.

Refactor Test to match BoardRepository.

-Add Test
	- findByCategoryAndCity Test
	- findByCategoryAndCityAndGu Test

LimNoah committed:
refactor: Refactor BoardController.

Refactor getBoardList API's parameter, name, return type.

LimNoah committed:
refactor: Refactor BoardController.

Refactor getBoardList API's name, parameter, return type.

LimNoah committed:
refactor: Refactor BoardService.

Refactor BoardService's methods.

- findByCategory
	- Change return type.
	- Refactor logic.
- boardPageToBoardResponseDtoPage
	- Change method name.
- boardToBoardResponseDto
	- Change method name.

LimNoah committed:
refactor: Refactor BoardService.

LimNoah committed:
refactor: Update BoardResponseDto's constructor.

Add constructor for use to change Board Entity to BoardResponseDto.

LimNoah committed:
refactor: Refactor BoardService.

Remove boardToBoardResponseDto method.

Because BoardResponseDto's constructor is created.

LimNoah committed:
refactor: Refactor BoardService.

LimNoah committed:
conf: Add data.sql

LimNoah committed:
refactor: Refactor BoardController

Change getBoardList's return method name.

LimNoah committed:
refactor: Refactor BoardControllerTest.

Add getBoardListNoCityAndGu test and getBoardList test.

LimNoah committed:
refactor: Refactor BoardService.

LimNoah committed:
test: Add BoardServiceTest.

- Test method
	- findByCategoryAndAddress()
	- findByCategoryAndCityAndGu()
- Given method
	- givenBoardList()
	- givenBoardPage()

LimNoah committed:
refactor: Refactor BoardService.

Change method name.

LimNoah committed:
test: Add BoardControllerTest.

- Use SpringBootTest annotation
- getBoardListNoCityAndGu test
	- URL에 주소 파라미터가 없을 때 실행되는 테스트.
- getBoardList test
	- URL에 시, 구 주소 파라미터 전부 있을 때 테스트.

LimNoah committed:
Merge branch 'feature/getBoardAPI' into develop

LimNoah committed:
refactor: Refactor BoardRepositoryTest.

LimNoah committed:
conf: Create data.sql for test.

LimNoah committed:
refactor: Refactor Member entity.

Add @Enumerated annotation.

LimNoah committed:
refactor: Refactor Board entity.

Add @Enumerated annotation.

LimNoah committed:
conf: Create data.sql for repository test.

LimNoah committed:
refactor: Refactor BoardRepositoryTest.

LimNoah committed:
refactor: Refactor Gender enum class.

MEN -> MALE
WOMEN -> FEMALE

LimNoah committed:
feat: Create createBoard method.

Create createBoard method for createBoardAPI.

LimNoah committed:
feat: Create BoardRequestDto class.

Create BoardRequestDto class for use createBoard API's parameter.

LimNoah committed:
test: Add createBoard test.

Add givenMember(), givenBoard() method for test.

LimNoah committed:
refactor: Refactor MemberRepositoryTest.

LimNoah committed:
test: Add createBoard test for createBoardAPI.

Add createdBoard test for createBoardAPI.

LimNoah committed:
feat: Add createBoard API.

LimNoah committed:
Merge branch 'feature/saveBoardAPI' into develop

LimNoah committed:
feat: Add deleteBoardById method to BoardService.

Add deleteBoardById() to use deleteBoardAPI.

LimNoah committed:
test: Add deleteBoard tests.

Add deleteBoard, deleteBoard_ResourcesNotFound tests.

LimNoah committed:
feat: Add deleteBoard API.

Add deleteBoard API to BoardController.

LimNoah committed:
feat: Add ResponseStatus annotation.

Add @ResponseStatus(HttpStatus.NOT_FOUND) annotation for test.

LimNoah committed:
test: Add deleteBoardAPI's test method.

- Add test method
	- deleteBoard() => success deleteBoard test
	- deleteBoard_ResourcesNotFound() => fail deleteBoard test

LimNoah committed:
Merge branch 'feature/deleteBoardAPI' into develop

LimNoah committed:
conf: Update build.gradle

Add dependencies
	- modelmapper
	- hateoas
	- validation starter

LimNoah committed:
refactor: Refactor BoareRequestDto.

LimNoah committed:
feat: Add updateBoard() for use updateBoardAPI.

Add updateBoard() for use updateBoardAPI

LimNoah committed:
test: Add updateBoard for test updateBoardAPI.

Add updateBoard test.

LimNoah committed:
feat: Add updateBoard API.

Add updateBoard API to BoardController.

LimNoah committed:
test: Add updateBoard test for updateBoardAPI.

Add updateBoard test.

LimNoah committed:
Merge branch 'feature/updateBoardAPI' into develop

LimNoah committed:
feat: Merge branch 'develop' of pmorigin.

LimNoah committed:
Merge branch 'develop' of https://github.com/bong-dari/bongdari-backend into develop

LimNoah committed:
conf: Delete application.yml.

tezla-lee commented on src/main/java/com/usbridge/bongdari/controller/dto/BoardRequestDto.java:
```diff
@@ -0,0 +1,43 @@
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class BoardRequestDto {
+    private Long id;
+    private int capacity;
+    private String contact;
+    private LocalDate startDate;
+    private LocalDate endDate;
+    private String details;
+    private String city;
+    private String gu;
+    private Category category;
```
추후에 validation 추가하심 될 듯

LimNoah replied:
넵 아직 제대로된 틀이 프론트분들이랑 얘기가 안되있어서 얘기 해보고 추가하겠습니다.

tezla-lee commented on src/test/java/com/usbridge/bongdari/repository/BoardRepositoryTest.java:
```diff
@@ -0,0 +1,49 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Transactional
```
DataJpaTest 안에 포함되어 있어서 지우시면 됨

LimNoah replied:
넵 확인했습니다.

tezla-lee commented on src/main/java/com/usbridge/bongdari/service/BoardService.java:
```diff
@@ -1,7 +1,81 @@
+            dtoList.add(new BoardResponseDto(board));
+        }
+
+        int start = (int)pageable.getOffset();
+        int end = Math.min((start + pageable.getPageSize()), dtoList.size());
+        return new PageImpl<>(dtoList.subList(start, end), pageable, dtoList.size());
+    }
```
subList 로 잘라주셨는데, 이미 데이터베이스에서 해당 페이지에 해당하는 데이터만 가져왔을거에요.

LimNoah replied:
아 그러네요 블로그 보고 따라해봐서 자세히 못봤습니다 dtoList로만 받아도 잘 되네요 수정하겠습니다 :)

tezla-lee commented on src/main/java/com/usbridge/bongdari/service/BoardService.java:
```diff
@@ -1,7 +1,81 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@Transactional
```
트랜잭셔널 어노테이션은 메소드 단위로 필요한 곳에만 다는 것이 좋다고 합니다.

LimNoah replied:
넵 알겠습니다 정보 감사요 :)

tezla-lee reviewed (changes requested):
확인해주세요~

LimNoah committed:
test: Update BoareRepositoryTest.java

Remove @Transactional annotation because @DataJpaTest already include @Transactional.

LimNoah committed:
refactor: Update BoardService.java

Change boardPageToBoardResponseDtoPage method's return parameter.

LimNoah committed:
refactor: Update BoardService.java

Move @Transactional annotation's location class area to method area.

LimNoah committed:
refactor: Update BoardService.java

Move @Transactional annotation's location class area to method area.

LimNoah reviewed (commented):
수정할 부분 수정하였습니다 확인 부탁드립니다.
